Util/+IMOS/gen_variables.m

```diff
@@ -0,0 +1,158 @@
+function [variables] = gen_variables(dimensions, v_names, v_types, v_data, varargin)
+% function [variables] = gen_variables(dimensions,v_names, v_types,v_data,varargin)
+%
+% Generate a toolbox variable cell of structs. Empty or incomplete
+% arguments will trigger random (names/types) or empty entries (data).
```

Comment:
no this also abstracts a bigger assignment pattern, do templating, and the most important thing for variables creation : automatic discovery of dimensions based on array size (all with proper stacktrace in case of error).

```
sample_data.variables{1}.dimensions     = [];
sample_data.variables{1}.name           = 'LATITUDE';
sample_data.variables{1}.typeCastFunc   = sr2func(netcdf3ToMatlabType(imosParameters(sample_data.variables{1}.name, 'type')));
... # all the other stuff 
sample_data.variables{3}.dimensions     = [];
sample_data.variables{3}.name           = 'NOMINAL_DEPTH';
sample_data.variables{3}.typeCastFunc   = str2func(netcdf3ToMatlabType(imosParameters(sample_data.variables{3}.name, 'type')));
sample_data.variables{3}.data           = sample_data.variables{3}.typeCastFunc(NaN);
sample_data.variables{end+1}.name       = 'TEMP';
sample_data.variables{end}.typeCastFunc = str2func(netcdf3ToMatlabType(imosParameters(sample_data.variables{end}.name, 'type')));
sample_data.variables{end}.dimensions   = 1;
sample_data.variables{end}.data         = sample_data.variables{end}.typeCastFunc(temp);
sample_data.variables{end}.coordinates  = coordinates;
...
```

all of the above is simplified to a way more semantical 4 lines:
```
dimensions = IMOS.gen_dimensions('timeSeries');
v_names = my_varnames; 
v_types = my_vartypes;
v_data = my_varcontent;
sample_data.variables = IMOS.gen_variables(dimensions,v_names,v_types,v_data);
```